src/main/java/somethingrandom/usecase/search/SearchItemsResult.java

```diff
@@ -0,0 +1,84 @@
+package somethingrandom.usecase.search;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+
+public class SearchItemsResult {
+    private final String name;
+    private final UUID uuid;
+    private @Nullable Instant couldBeReadyWhen = null;
+
+    /**
+     * Creates a new search result containing just the name.
+     * <p>
+     * This is primarily intended for use with reference items.
+     *
+     * @param name The name of the item.
+     * @param uuid The identifier of the item.
+     */
+    SearchItemsResult(String name, UUID uuid) {
+        this.name = name;
+        this.uuid = uuid;
+    }
+
+    /**
+     * Creates a new search result containing the name, and a duration.
+     * <p>
+     * In addition, the time the query was submitted should be provided. This
+     * ensures that differences in time getting results don't change how they
+     * are ordered.
+     * <p>
+     * This is primarily intended for use with actionable items.
+     *
+     * @param name The name of the item.
+     * @param querySubmitted The time when the use-case was executed.
+     * @param duration How long the item is expected to complete.
+     */
+    SearchItemsResult(String name, UUID uuid, Instant querySubmitted, Duration duration) {
```

Comment:
This is a little late to note, but does this mean users will have to type the exact name of the item they're searching for? Since we originally had the output as a collection of all the possible results?

Reply:
To clarify, the current structure of this output data entails we would be retrieving/ presenting exactly one item, which doesn't match up with the output boundary I don't think

Reply:
It's still a collection of results, so there can be a bunch of them! (I could've had the output data contain a list, but then we'd need more classes for basically nothing.)

It'll currently list all results that contain the query.